Source/DafnyCore/Util.cs

```diff
@@ -845,7 +845,8 @@ public class ExpressionTester {
 
       if (expr is IdentifierExpr expression) {
         if (expression.Var != null && expression.Var.IsGhost) {
-          reporter?.Error(MessageSource.Resolver, expression, "a ghost variable is allowed only in specification contexts");
+          reporter?.Error(MessageSource.Resolver, expression,
+              $"ghost variables such as {expression.Name} are allowed only in specification contexts. {expression.Name} was inferred to be ghost based on its use or declaration.");
```

Comment:
Is ghostness actually inferred based on use?  I thought it was only from the body of the variable, or from it being marked "ghost" explicitly.

Reply:
By use here I did mean its initialization, but I was not sure if more of its use was considered.
Let me know if you think a wording change is needed (it will require updating a fair number of tests)

Reply:
Yes, I think it would be better to not mention the use.  I would just remove the second half of the message.

Reply:
Btw, for these sort of edits, I would:

1. Use `git format-patch HEAD~1` to produce a patch
2. Open the patch (called `0001-….patch`) in a text editor
3. Use search and replace, with a regexp, to apply the change to all files
4. Use `git am` to apply the patch

This way there's no need to edit all the files by hand.
Alternatively, `runTests.py` can regenerate these files automatically for you.

Reply:
I reworded, but the point of the issue was to be more explanatory.
sed did this simple change quite readily